engine/src/test/java/com/redhat/cloud/notifications/processors/webhook/WebhookTest.java

```diff
@@ -146,6 +176,75 @@ private void testRetry(boolean shouldSucceedEventually) {
         }
     }
 
+    @Test
+    void testDisableEndpointOnClientError() {
+        featureFlipper.setDisableWebhookEndpointsOnFailure(true);
+
+        HttpRequest mockServerRequest = getMockHttpRequest("/client-error", request -> response().withStatusCode(401));
+        try {
+            Action action = buildWebhookAction();
+            Event event = new Event();
+            event.setAction(action);
+            Endpoint ep = buildWebhookEndpoint(getMockServerUrl() + "/client-error");
+            persistEndpoint(ep);
+            assertTrue(ep.isEnabled());
+            statelessSessionFactory.withSession(statelessSession -> {
+                webhookTypeProcessor.process(event, List.of(ep));
+                micrometerAssertionHelper.assertCounterIncrement(DISABLED_ENDPOINTS_COUNTER, 1, ERROR_TYPE_TAG_KEY, CLIENT_TAG_VALUE);
+                assertFalse(getEndpoint(ep.getId()).isEnabled());
+            });
+        } finally {
+            // Remove expectations
+            MockServerLifecycleManager.getClient().clear(mockServerRequest);
+        }
+
+        featureFlipper.setDisableWebhookEndpointsOnFailure(false);
+    }
+
+    @Test
+    void testDisableEndpointOnServerError() {
+        featureFlipper.setDisableWebhookEndpointsOnFailure(true);
+
+        HttpRequest mockServerRequest = getMockHttpRequest("/server-error", request -> response().withStatusCode(503));
+        try {
+            Action action = buildWebhookAction();
+            Event event = new Event();
+            event.setAction(action);
+            Endpoint ep = buildWebhookEndpoint(getMockServerUrl() + "/server-error");
+            persistEndpoint(ep);
+            assertTrue(ep.isEnabled());
+            statelessSessionFactory.withSession(statelessSession -> {
+                for (int i = 0; i < 4; i++) {
+                    /*
+                     * The processor retries 3 times in case of server error,
+                     * so the endpoint will actually be called 12 times in this test.
```

Comment:
Shouldn't it only be called it 10 times ? Or do we keep trying even if the endpoint was disabled by this?

Reply:
Good question. With the current code, Failsafe will apply its retry policy until it's done retrying a particular call (3 times by default). We could probably interrupt the Failsafe retries somehow but in the meantime it doesn't really hurt if we don't. There is one unlikely yet possible bad scenario: if call #10 fails and call #11 succeeds, we will send an email while the endpoint will actually not be disabled. Maybe that a good enough reason to not let Failsafe complete its retry cycle.

Reply:
Lets see how this behaves

Reply:
We could create a JIRA to follow up this particular issue